### ISSUE TITLE
ConfirmDialog：confirmDialog method in unstyled mode, the pop-up window style not apply

### ISSUE THREAD

**Issue by shenpvip** [primefaces/primereact#5656]
### Describe the bug

confirmDialog method in unstyled mode, the pop-up window style does not apply

### Reproducer

https://stackblitz.com/edit/react-hzzvhp?file=src%2Findex.js,src%2FApp.js

### PrimeReact version

10.2.1

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

**Comment by kl-nevermore** [comment]
Should confirmProps and props be merged?
like...
```tsx

const getCurrentProps = () => Object.assign(props,confirmProps.current)
const getPropValue = (key) => Object.assign(props,confirmProps.current)[key];
```

**Comment by melloware** [comment]
@kl-nevermore is this still happening in the main branch?  I thought I had fixed this before when someone reported it but i could be wrong?

**Comment by kl-nevermore** [comment]
@melloware 
I just tested in main branch
It seems like only the buttons work
![image](https://github.com/primefaces/primereact/assets/25167742/72f9ccc8-5dff-4d70-ae45-3870e13ee3d6)

**Comment by kl-nevermore** [comment]
After merging props, it can work, but I'm not sure if it's the correct solution

**Comment by melloware** [comment]
well submit a PR please I can take a look.

**Comment by kl-nevermore** [comment]
Still can not
![image](https://github.com/primefaces/primereact/assets/25167742/f2aceb20-3864-426f-80b3-617dc797deac)

**Comment by kl-nevermore** [comment]
1. the pt passed to root is currentProps.pt
2. the `currentProps` from `getCurrentProps`
3. getCurrentProps return confirmProps.current || props
4. confirmProps.current from confirm's arguments
5. when invoke confirmDialog,the confirmDialog's arguments as props
6. lost the component's props?

**Comment by melloware** [comment]
@kl-nevermore i just tried your sandbox with 10.3.0 and it has the same issue?

https://stackblitz.com/edit/react-hzzvhp-6btcgt?file=src%2Findex.js,src%2FApp.js

**Comment by melloware** [comment]
Had to revert this fix it broke the showcase.

**Comment by kl-nevermore** [comment]
@melloware 
I just tested use tag 10.2.1, I guess it had some impact.


![20240106095350_rec_](https://github.com/primefaces/primereact/assets/25167742/af5f3929-5896-43c0-b8a1-43e1dacec424)

**Comment by melloware** [comment]
OK well if you try with the current showcase you will see the issue I am seeing if you want to take another look!

**Comment by kl-nevermore** [comment]
I found #5401 was rolled back and so were many changes

**Comment by melloware** [comment]
@kl-nevermore let's revisit this now that the main branch is up to date.

**Comment by kl-nevermore** [comment]
@melloware 
I'll follow up today

**Comment by kl-nevermore** [comment]
like unstyled, passed `props.pt` to `Dialog`
```diff
const rootProps = mergeProps(
    {
        visible: visibleState,
        className: classNames(getPropValue('className'), cx('root')),
        footer,
        onHide: hide,
        breakpoints: getPropValue('breakpoints'),
+       pt: props.pt,
-       pt: currentProps.pt
        unstyled: props.unstyled,
        appendTo: getPropValue('appendTo'),
        __parentMetadata: {
            parent: metaData
        }
    },
    ConfirmDialogBase.getOtherProps(currentProps)
);

```

**Comment by melloware** [comment]
I thought only one confirm dialog could be on the screen at once?   But I guess the could be stacked but from a UI perspective only 1 confirm dialog at a time makes more sense to me.

**Comment by melloware** [comment]
The latest fixes have been applied in 10.5.1
